src/hotspot/share/opto/escape.cpp

```diff
@@ -3739,6 +3773,34 @@ void ConnectionGraph::dump(GrowableArray<PointsToNode*>& ptnodes_worklist) {
   }
 }
 
+void ConnectionGraph::print_statistics() {
+  tty->print("No escape = %d, Arg escape = %d, Global escape = %d", Atomic::load(&_no_escape_counter), Atomic::load(&_arg_escape_counter), Atomic::load(&_global_escape_counter));
+  tty->print_cr(" (EA executed in %7.2f seconds)", Atomic::load(&_time_elapsed) * 0.001);
+}
+
+void ConnectionGraph::escape_state_statistics(GrowableArray<JavaObjectNode*>& java_objects_worklist) {
+  _compile->_local_no_escape_ctr = 0;
+  _compile->_local_arg_escape_ctr = 0;
+  _compile->_local_global_escape_ctr = 0;
```

Comment:
I don't think you need these `Compile` counters - make them local to update corresponding `ConnectionGraph` counters after loop. As I said these counters are not accurate anyway. Unless you want to track for which allocation which counter was recorded. Which I think will be over-kill because the most EA done with one iteration.